pkg/controller/worker/machines.go

```diff
@@ -324,3 +328,27 @@ func computeAdditionalHashData(pool extensionsv1alpha1.WorkerPool, workerConfig
 
 	return additionalData
 }
+
+func readMachineConfiguration(pool extensionsv1alpha1.WorkerPool) machinev1alpha1.MachineConfiguration {
```

Comment:
This can be a helper function that you can put to `gardener/gardener/extensions/pkg/controller/worker/genericactuator` package. Then you don't need to copy this code for every extension.
Do you want to add this in https://github.com/gardener/gardener/pull/2563?

Reply:
That's a great suggestion, I'll do that. Thanks.

Reply:
I made the change at #2563 https://github.com/gardener/gardener/commit/47b627f032934b69ee2c97963d3073f83cf6ae96#diff-ce16933f7218046c410bbaed72f771b7R573